x/poolmanager/create_pool.go

```diff
@@ -53,40 +52,53 @@ func (k Keeper) CreatePool(ctx sdk.Context, msg types.CreatePoolMsg) (uint64, er
 		return 0, err
 	}
 
+	// Send initial liquidity to the pool's address.
+	initialPoolLiquidity := msg.InitialLiquidity()
+	err = k.bankKeeper.SendCoins(ctx, sender, pool.GetAddress(), initialPoolLiquidity)
+	if err != nil {
+		return 0, err
+	}
+
+	return pool.GetId(), nil
+}
+
+// CreatePoolZeroLiquidityNoFee creates a pool from given message without sending any initial liquidity to the pool
+// and paying a fee. This is meant to be used for creating the pools internally. For example, in the upgrade handler.
+// This is not meant to be used for creating pools from user messages.
+func (k Keeper) CreatePoolZeroLiquidityNoFee(ctx sdk.Context, msg types.CreatePoolMsg) (types.PoolI, error) {
```

Comment:
Can we rename this? Maybe "NoCreationFee"? When I first read it I thought it meant zero swap fee